### PR TITLE
minor fixes to allow dual numbers to propgate through

### PR DESCRIPTION
**Pull request by andrewning**
The one limitation of what I've done is I'm assuming the dual numbers are contained in ydata rather than xdata (line 42).  That would be the normal case.  I have a hard time imaging a scenario where one would want derivatives of xdata without the ydata changing.  If we wanted to allow that we could change that line to a unior type between xdata and ydata.

### CODE REVIEW

**Comment by taylormcd** [comment]
We should probably add a simple dual number test.

File: src/interpolate.jl [hunk 1]
```diff
@@ -8,14 +8,14 @@ using OffsetArrays: OffsetVector
 ## ------ Akima Interpolation  ---------
 
 # struct used internally
-struct Akima{TF, T1<:AbstractVector{TF}, T2<:AbstractVector{TF}}
+struct Akima{TF1, TF2, T1<:AbstractVector{TF1}, T2<:AbstractVector{TF2}}
```
Comment (taylormcd) [inline src/interpolate.jl]: This could be a breaking change if anyone references Akima as Akima{TF, T1, T2}.

Comment (taylormcd) [reply src/interpolate.jl]: I'm fine with changing this though.

Comment (andrewning) [reply src/interpolate.jl]: yeah I guess technically breaking, but none of us are using it like that, and I don't see how to avoid it.

**Comment by andrewning** [comment]
generalized so you could get derivatives of xdata and/or ydata, added the unit tests, and added test dependencies since running the test script now requires ForwardDiff.

**Comment by taylormcd** [comment]
I moved the test dependencies to the main Project.toml.  I also simplified the Akima type to reduce the number of parameters.

**Comment by andrewning** [comment]
- I don't think that is type stable, but I haven't checked.
- The dependencies shouldn't be in the main Project.toml.  That's are in the test/Project.toml because they are test-specific dependencies.  This assumes >= julia 1.2 but we decided to only support recent versions of julia anyway.

**Comment by andrewning** [comment]
I reverted your commit on Project.toml but kept your commit for type changes (it is type stable).  It does simplify the Akima type but then complicates the function signature for Akima so not sure it's a net win or not.  I don't think it is worth worrying about so I'll just keep it.

**Pull request merged**